apps/web/src/design-system/table/Table.tsx

```diff
@@ -38,16 +50,21 @@ export interface ITableProps<T extends object> {
   data?: T[];
   loading?: boolean;
   pagination?: any;
-  onRowClick?: (row: Row<T>) => void;
   noDataPlaceholder?: React.ReactNode;
   loadingItems?: number;
   hasMore?: boolean;
   minimalPagination?: boolean;
+  withSelection?: boolean;
+  withRowClickSelection?: boolean;
+  initialSelectedRows?: Record<IdType<T>, boolean>;
+  onRowClick?: (row: Row<T>) => void;
+  onRowSelect?: (row: Row<T>) => void;
```

Comment:
Added the functionality to the table that will allow us to have a radio button in the first column of the table. Only one row could be selected and a callback triggered on selection.